packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts

```diff
@@ -169,7 +169,12 @@ export function createMikrORMEntity() {
       Enum({
         items: () => field.dataType.options!.choices,
         nullable: field.nullable,
-        default: field.defaultValue,
+        /**
+         * MikroORM does not ignore undefined values for default when generating
+         * the database schema SQL. Conditionally add it here to prevent undefined
+         * from being set as default value in SQL.
+         */
+        ...(field.defaultValue && { default: field.defaultValue }),
```

Comment:
todo: if the defaultValue is a false or null or '' then this wont work. Is would suggest to use the isDefined util here wdyt?

Reply:
good catch!